### ISSUE TITLE
Requesting HelpHelper / @Html.Checkbox etc 

### ISSUE THREAD

**Issue by alexellis** [kirbysayshi/vash#57]
Hi,

One of the nice features of Razor is the @Html helper for CheckboxFor etc..

Would you consider implementing a basic / initial version of this?

Thanks

**Comment by kirbysayshi** [comment]
I probably won't (vash's api is alread larger than I want it to be), but the helper system is meant to be extensible by developers. If you attach a function to vash.helpers, it will be accessible within the executing template context.

**Comment by alexellis** [comment]
Can you give an example please? Also is Vash still being maintained and developed?

**Comment by kirbysayshi** [comment]
yes, vash is still being maintained. Development has slowed down though, mostly because it's relatively stable and the needs of the community, in general, are moving towards things like React.

It's been awhile since I used actual ASP MVC, but I believe this would do what you want (client side usage is similar, but I'm not sure how you're deploying/executing code):

```js
var helpers = require('vash/runtime').helpers;
helpers.CheckboxFor = function(checked, attributes) {
  this.buffer.push('<input type="checkbox" ' + (checked ? 'checked' : '')
    + Object.keys(attributes).map(function(attribute) {
      return attribute + '=' + this.escape(attributes[attribute]);
    })
    + '>'
  ); 
}
```

You can also use a [compiled helper](https://github.com/kirbysayshi/vash#compiled-helpers-), or use a function that returns a string. The choice is up to you, depending on your needs.

**Comment by alexellis** [comment]
I will try this out. I didn't understand the reference to React though - having tried it briefly in a hackathon event it does not seem to make any sense for node.js templating at the server-side. The benefits of a 'virtual dom' don't really seem to exist.

It would be great to think vash could continue to be supported through future versions of node etc. Failing that - handlebars seems to be very limited in comparison (particularly when it comes to conditional statements) - perhaps people would migrate to EJS?

**Comment by kirbysayshi** [comment]
Don't worry, I'll continue to support vash! I'm not keen on adding to the API / maintenance too much though, because it's already a relatively large project in terms of API surface area. So if you'd like helpers and things, it's probably best to create a separate package for those that you require in your app.

What I meant about react is that it's a very stable templating solution that can use the templates on both the server and client. While you're right that a virtual dom isn't useful on the server, the main benefit is actually just the large community around JSX: tools, syntax highlighting, etc. It's pretty easy to take a React component and render it to a string: https://facebook.github.io/react/docs/top-level-api.html#reactdomserver.rendertostring.
